adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/authorization/PolicyEnforcer.java

```diff
@@ -263,20 +267,35 @@ public PathConfig matches(String targetUri) {
             pathConfig = super.matches(targetUri);
 
             if (enforcerConfig.getLazyLoadPaths() || enforcerConfig.getPathCacheConfig() != null) {
```

Comment:
Why are we doing lazy loading even if lazy loading is disabled and cache is enabled?: `|| enforcerConfig.getPathCacheConfig() != null`?

Reply:
To be honest, I don't know the exact reason. This is also something I would like to check and possibly remove so that we only fetch resources if lazy load is enabled. There must be some use case in test that relies on it.

We can create a JIRA and look at that more closely.